examples/step-79/step-79.cc

```diff
@@ -0,0 +1,2444 @@
+    std::vector<double> old_upper_slack_multiplier_values(n_q_points);
+    std::vector<double> old_lower_slack_values(n_q_points);
+    std::vector<double> old_upper_slack_values(n_q_points);
+    std::vector<double> old_unfiltered_density_values(n_q_points);
+    std::vector<double> old_unfiltered_density_multiplier_values(n_q_points);
+    std::vector<double> filtered_unfiltered_density_values(n_q_points);
+    std::vector<double> filter_adjoint_unfiltered_density_multiplier_values(
+      n_q_points);
+
+    using namespace ValueExtractors;
+    for (const auto &cell : dof_handler.active_cell_iterators())
+      {
+        cell_rhs = 0;
+
+        cell->get_dof_indices(local_dof_indices);
+
+        fe_values.reinit(cell);
+
+        lambda.value_list(fe_values.get_quadrature_points(), lambda_values);
+        mu.value_list(fe_values.get_quadrature_points(), mu_values);
+
+        fe_values[densities<dim>].get_function_values(test_solution,
+                                                      old_density_values);
+        fe_values[displacements<dim>].get_function_values(
+          test_solution, old_displacement_values);
+        fe_values[displacements<dim>].get_function_divergences(
+          test_solution, old_displacement_divs);
+        fe_values[displacements<dim>].get_function_symmetric_gradients(
+          test_solution, old_displacement_symmgrads);
+        fe_values[displacement_multipliers<dim>].get_function_values(
+          test_solution, old_displacement_multiplier_values);
+        fe_values[displacement_multipliers<dim>].get_function_divergences(
+          test_solution, old_displacement_multiplier_divs);
+        fe_values[displacement_multipliers<dim>]
+          .get_function_symmetric_gradients(
+            test_solution, old_displacement_multiplier_symmgrads);
+        fe_values[density_lower_slacks<dim>].get_function_values(
+          test_solution, old_lower_slack_values);
+        fe_values[density_lower_slack_multipliers<dim>].get_function_values(
+          test_solution, old_lower_slack_multiplier_values);
+        fe_values[density_upper_slacks<dim>].get_function_values(
+          test_solution, old_upper_slack_values);
+        fe_values[density_upper_slack_multipliers<dim>].get_function_values(
+          test_solution, old_upper_slack_multiplier_values);
+        fe_values[unfiltered_densities<dim>].get_function_values(
+          test_solution, old_unfiltered_density_values);
+        fe_values[unfiltered_density_multipliers<dim>].get_function_values(
+          test_solution, old_unfiltered_density_multiplier_values);
+        fe_values[unfiltered_densities<dim>].get_function_values(
+          filtered_unfiltered_density_solution,
+          filtered_unfiltered_density_values);
+        fe_values[unfiltered_density_multipliers<dim>].get_function_values(
+          filter_adjoint_unfiltered_density_multiplier_solution,
+          filter_adjoint_unfiltered_density_multiplier_values);
```

Comment:
This looks similar to what you were doing above in that huge block of code. Would it make sense to put this into a function?